systemtest/src/test/java/io/strimzi/systemtest/specific/RackAwarenessST.java

```diff
@@ -0,0 +1,206 @@
+        String kafkaClusterTargetName = clusterName + "-target";
+
+        resourceManager.createResource(extensionContext,
+                KafkaTemplates.kafkaEphemeral(kafkaClusterSourceName, 1, 1).build());
+        resourceManager.createResource(extensionContext,
+                KafkaTemplates.kafkaEphemeral(kafkaClusterTargetName, 1, 1).build());
+
+        resourceManager.createResource(extensionContext,
+                KafkaMirrorMaker2Templates.kafkaMirrorMaker2(clusterName, kafkaClusterTargetName, kafkaClusterSourceName, 1, false)
+                        .editSpec()
+                            .withNewRack(TOPOLOGY_KEY)
+                        .endSpec()
+                        .build());
+
+        LOGGER.info("MirrorMaker2 cluster deployed successfully");
+        String deployName = KafkaMirrorMaker2Resources.deploymentName(clusterName);
+        String podName = PodUtils.getPodNameByPrefix(namespace, deployName);
+        Pod pod = kubeClient().getPod(namespace, podName);
+
+        // check that spec matches the actual pod configuration
+        Affinity specAffinity = kubeClient().getDeployment(namespace, deployName).getSpec().getTemplate().getSpec().getAffinity();
+        NodeSelectorRequirement specNodeRequirement = specAffinity.getNodeAffinity().getRequiredDuringSchedulingIgnoredDuringExecution().getNodeSelectorTerms().get(0).getMatchExpressions().get(0);
+        NodeAffinity podAffinity = pod.getSpec().getAffinity().getNodeAffinity();
+        NodeSelectorRequirement podNodeRequirement = podAffinity.getRequiredDuringSchedulingIgnoredDuringExecution().getNodeSelectorTerms().get(0).getMatchExpressions().get(0);
+        assertThat(podNodeRequirement, is(specNodeRequirement));
+        assertThat(podNodeRequirement.getKey(), is(TOPOLOGY_KEY));
+        assertThat(podNodeRequirement.getOperator(), is("Exists"));
+
+        // check Kafka client rack awareness configuration
+        String podNodeName = pod.getSpec().getNodeName();
+        String hostname = podNodeName.contains(".") ? podNodeName.substring(0, podNodeName.indexOf(".")) : podNodeName;
+        String commandOut = cmdKubeClient(namespace).execInPod(podName, "/bin/bash", "-c", "cat /tmp/strimzi-connect.properties | grep consumer.client.rack").out().trim();
+        assertThat(commandOut.equals("consumer.client.rack=" + hostname), is(true));
+    }
+
+    @BeforeEach
+    void createTestResources(ExtensionContext extensionContext) {
+        storageMap.put(extensionContext, new TestStorage(extensionContext));
+    }
+
+    @BeforeAll
+    void setUp() {
+        assumeFalse(Environment.isNamespaceRbacScope());
+        clusterOperator.unInstall();
+        clusterOperator = clusterOperator
+                .defaultInstallation()
+                .withOperationTimeout(CO_OPERATION_TIMEOUT_SHORT)
+                .withReconciliationInterval(Constants.RECONCILIATION_INTERVAL)
+                .createInstallation()
+                .runInstallation();
+    }
```

Comment:
Just a minor comment before my whole review. You cannot modify the configuration of the `shared` Cluster Operator when you label the whole test suite as `@ParallelSuite`. It should be `@IsolatedSuite` and the name has to be `RackAwarenessIsolatedST`.

Reply:
Correct, thanks @see-quick for spotting this. We may actually don't need these configurations, as in `testConnectRackAwareness` we can simply check that the pod is unschedulable with an invalid `topologyKey`, without waiting for the CO timeout. Let me do and test this change.

As a side note, the CO customization in parallel suites could be a common pitfall, especially when converting from isolated test suites (like in this case). I think we should prevent that from happening in the first place by raising a runtime error at the ST framework level. WDYT?

Reply:
> I think we should prevent that from happening in the first place by raising a runtime error at the ST framework level. WDYT?

Yes, I totally agree. Plus if there is no need for CO configuration, you can remove it completely because the shared CO for `@ParallelSuites` is deployed at the start of the execution in `BeforeAllOnce.class`.